lib/pgBackRest/Backup/Filter/PageChecksum.pm

```diff
@@ -0,0 +1,196 @@
+
+####################################################################################################################################
+# isLibC
+#
+# Does the C library exist?
+####################################################################################################################################
+sub isLibC
+{
+    return $bLibC;
+}
+
+push @EXPORT, qw(isLibC);
+
+####################################################################################################################################
+# CONSTRUCTOR
+####################################################################################################################################
+our @ISA = ();                                                      ## no critic (ClassHierarchies::ProhibitExplicitISA)
+
+sub new
+{
+    my $class = shift;
+
+    # Assign function parameters, defaults, and log debug info
+    my
+    (
+        $strOperation,
+        $self,
+        $iSegmentNo,
+        $iWalId,
+        $iWalOffset,
+    ) =
+        logDebugParam
+        (
+            __PACKAGE__ . '->new', \@_,
+            {name => 'self', trace => true},
+            {name => 'iSegmentNo', trace => true},
+            {name => 'iWalId', trace => true},
+            {name => 'iWalOffset', trace => true},
+        );
+
+    # Bless with the new class
+    @ISA = $self->isA();                                            ## no critic (ClassHierarchies::ProhibitExplicitISA)
+    bless $self, $class;
+
+    # Set variables
+    $self->{iSegmentNo} = $iSegmentNo;
+    $self->{iWalId} = $iWalId;
+    $self->{iWalOffset} = $iWalOffset;
+
+    # Create the result object
+    $self->{hResult}{bValid} = true;
+    $self->{hResult}{bAlign} = true;
+
+    # Return from function and log return values if any
+    return logDebugReturn
+    (
+        $strOperation,
+        {name => 'self', value => $self}
+    );
+}
+
+####################################################################################################################################
+# read - validate page checksums
+####################################################################################################################################
+sub read
+{
+    my $self = shift;
+    my $rtBuffer = shift;
+    my $iSize = shift;
+
+    # Call the io method
+    my $iActualSize = $self->SUPER::read($rtBuffer, $iSize);
+
+    # Validate page checksums for the read block
+    if ($iActualSize > 0)
+    {
+        # If the buffer is not divisible by 0 then it's not valid
+        if (!$self->{hResult}{bAlign} || ($iActualSize % PG_PAGE_SIZE != 0))
+        {
+            if (!$self->{hResult}{bAlign})
+            {
+                confess &log(ASSERT, "should not be possible to see two misaligned blocks in a row");
+            }
+
+            $self->{hResult}{bValid} = false;
+            $self->{hResult}{bAlign} = false;
+            delete($self->{hResult}{iyPageError});
+        }
+        elsif ($iActualSize > 0)
```

Comment:
Looks like a holdover from a previous version when actual size was not checked so early.  Fixed.